tests/lax_numpy_reducers_test.py

```diff
@@ -866,5 +866,18 @@ def testCumulativeSumBool(self):
                              dtype=jnp.bool_)
     np.testing.assert_array_equal(np.array([[True], [True], [False]]), out)
 
+  @jtu.sample_product(
+    [dtype for dtype in int_dtypes + unsigned_dtypes],
+  )
+  @jtu.ignore_warning(category=NumpyComplexWarning)
+  def test_low_bitwidth_sum(self, dtype) -> None:
+    rng = rng_factory(self.rng())
+    shape = (10, 10)
+    tol = 0.01
+    args_maker = lambda: [rng(shape, dtype)]
+    sum_jit = jax.jit(jnp.sum)
+    np_sum = lambda x: np.sum(x)
```

Comment:
I don't think `np.sum` implements the casting behavior we expect here, so we'll need the reference operation to cast to the expected type like in `jnp.sum`.

Reply:
According to the docstring in the jax implementation of the promotion numpy promotes to int64. I also added an explicit promotion to int32 in the test function to have at least the "JAX equivalent" promotion.

Reply:
NumPy's documented behavior only applies to types it knows about – in practice if you try a NumPy sum over an int2 array, it will not promote.